lib/find_non_repeated_letter.rb

```diff
@@ -9,5 +9,18 @@
 
 
 def find_first_non_repeated_letter(str)
-  #write your method here
+    ary = str.chars
+    x = 0
+    
+    while x <= ary.length
+        if ary.length <= 1
+            return ary[x]
+        elsif ary[x] == ary[x + 1] || ary[x + 1] == ary[x + 2]
```

Comment:
I found that I needed the secondary conditional because it would return the first different element even if the element following is the same as the latter. So with the string 'aabbccdeeffg', it would return the first 'b' as it is different when compared to the second 'a'.